src/types/list.md

```diff
@@ -1 +1,69 @@
+
+## Values
+
+We write lists as a sequence of integers, separated by commas, all enclosed with two square brackets. So then we have valid int lists as `[1, 2, 3]`, `[1, 5, 1, 5, 0]`, and `[]`, with the latter representing the empty list. We also refer to the empty list as _nil_ (in addition, you can type `nil` instead of `[]` in code). 
+
+The other essential component to lists is what is known as the `::` operator, referred to as "cons". Cons can be used as a constructor for any fixed type of list, so we say that for any given type `t`:
+```sml
+(op ::) : t * t list -> t list
+```
+
+Cons takes in a value `v` of type `t` and `t list`, and prepends `v` to the front of the given list. For ints in particular, we have that `1 :: [2, 3]` steps to `[1, 2, 3]`. Additionally, cons is _right-associative_. This means that in a continuous stream of applications of cons, they are evaluated from _right-to-left_. This means that `1 :: 2 :: 3 :: []` is implicitly denoting `1 :: (2 :: (3 :: []))`, as the calls to cons associate to the right. So, similarly to before, a `1 :: [2, 3]`, `[1, 2, 3]`, and `1 :: 2 :: 3 :: []` are exactly equivalent, and denote the same list.
+
+Cons is also a _constructor_, meaning that it can be used to pattern match and deconstruct lists. As such, we can write basic functions to compute the length of a list as follows:
+```sml
+(* length : int list -> int *)
+(* REQUIRES: true *)
+(* ENSURES: length L ==> the length of L *)
+fun length ([] : int list) : int = 0
+  | length (x::xs : int list) : int = 1 + length xs
+```
+Given a non-empty list, this function simply binds the first element of the list to the identifier `x`, discards it, and then recursively calls `length` to find the length of the remaining list `xs`, adding 1 to the result.
+
+The definition of an int list thus corresponds to:
+```sml
+datatype int list = [] | :: of int * int list
+```
+where an int list can either be the empty list `[]`, or it can be `::` of a first element and the rest of the list (where `::` is an infix operator, so instead of being written as `::(x, xs)`, we have `x :: xs`). Note that this is not actually valid syntax, but you can think of the definition of int lists in this way.
```

Comment:
```suggestion
where an `int list` can either be the empty list `[]`, or it can be `::` of a first element and the rest of the list (where `::` is an infix operator, so instead of being written as `:: (x,xs)`, we have `x :: xs`). Note that this is not actually valid syntax, but you can think of the definition of int lists in this way.
```
(Same change (int list to `int list`) in potentially other places also, but I noted the ones I saw)